src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -878,7 +878,30 @@ private static bool IsGlobalAttributeTarget(SyntaxToken token)
 
         private bool IsPossibleAttributeDeclaration()
         {
-            return this.CurrentToken.Kind == SyntaxKind.OpenBracketToken;
+            // Have to at least start with `[` to be an attribute
+            if (this.CurrentToken.Kind != SyntaxKind.OpenBracketToken)
+                return false;
+
+            using var _ = this.GetDisposableResetPoint(resetOnDispose: true);
```

Comment:
>using var _ = this.GetDisposableResetPoint(resetOnDispose: true); [](http://example.com/codeflow?start=12&length=65)

I am not going to block on this, but I would prefer if this was a `using` statement with a block. The chosen style has a big negative impact on code readability and, as a result, on code maintainability as well, in my opinion.

Reply:
I find it readable.  But if you prefer a block, i'll move to that.  

> In general, we care more about clarity of the code vs. brevity and lack of extra indentation.

I care about that as well.  But i find this form much more readable.  I don't need to figure out where it ends.  I know where it ends (since it extends to the end of the containing block.  So i can instantly know that any return within it resets back to a correct place.  A using with a block has to then be examined to ensure nothing happens outside that block.  This was a core benefit to this form in the first place (esp. as that's the common case for usings and try/finalys).

Reply:
switched.

Reply:
> I know where it ends (since it extends to the end of the containing block.  

You wrote the code, of course you know that now.

One of the problems is that we don't know where it begins, we don't even see that there is a `using` there, because it looks very much as a regular local declaration. We want things to be explicit, we want them to be in our face. If there is a context, we want it to be obvious.

Reply:
> You wrote the code, of course you know that now.

I know that now, and in the future.  And that's *very much* the semantics i want.  :)

With a block-style using, we've def seen people accidentally add the code *after* teh block.  This requires *much* more extensive review to ensure proper semantics.  With variable-style-using, i don't have to worry about such accidents.  This is very beneficial in my eyes or readability and maintainability.  YMMV though.  I've made the change since i don't feel strongly about it, and i'm happy to have this code match your style.

I'm just pointing out that i acutally prefer the other form not for style, but because i have found it leads to more correct code overall.

>  We want things to be explicit, we want them to be in our face.

It is explicit.  There is nothing implicit here.  As to `in our face`, that's going to be up to the individual.  I personally find `using` to be so `in your face` as to be unavoidable to miss.